Formula/petsc.rb

```diff
@@ -0,0 +1,40 @@
+class Petsc < Formula
+  desc "Tools for solving partial differential equations in parallel"
+  homepage "https://www.mcs.anl.gov/petsc/"
+  url "http://ftp.mcs.anl.gov/pub/petsc/release-snapshots/petsc-lite-3.9.0.tar.gz"
+  sha256 "a233e0d7f69c98504a1c3548162c6024f7797dde5556b83b0f98ce7326251ca1"
+
+  depends_on "hdf5"
+  depends_on "hwloc"
+  depends_on "metis"
+  depends_on "netcdf"
+  depends_on "open-mpi"
+  depends_on "scalapack"
+  depends_on "suite-sparse"
+
+  def install
+    ENV["CC"] = "mpicc"
+    ENV["CXX"] = "mpicxx"
+    ENV["F77"] = "mpif77"
+    ENV["FC"] = "mpif90"
+    system "./configure", "--prefix=#{prefix}",
+                          "--with-debugging=0",
+                          "--with-scalar-type=real",
+                          "--with-x=0"
+    system "make", "all"
+    system "make", "install"
+  end
+
+  test do
+    test_case = "#{pkgshare}/examples/src/ksp/ksp/examples/tutorials/ex1.c"
+    system "mpicc", test_case, "-I#{include}", "-L#{lib}", "-lpetsc", "-o", "test"
+    output = shell_output("./test")
+    # This PETSc example prints several lines of output. The last line contains
+    # an error norm, expected to be small.
+    line = output.lines.last
+    match = /(?<=^Norm of error )(.+)(?=,)/.match(line)
```

Comment:
@ilovezfs Parsing the text seems more robust (against potential output format changes) than checking the number of tokens on the last line. It's obviously not ironclad. I'm happy to change back if preferred.

Reply:
You should be able to use `assert_match`